fn/option.go

```diff
@@ -58,6 +58,12 @@ func ElimOption[A, B any](o Option[A], b func() B, f func(A) B) B {
 	return b()
 }
 
+// Unwrap is used to extract the value from an option. If the option is empty,
+// then the zero value of the type that parameterizes the option is returned.
+func (o Option[A]) Unwrap() (A, bool) {
```

Comment:
Makes it easier to do unsafe access (eg: ignoring the second arg), but we basically already have that exposed with `UnwrapToPtr` as is 🤷

Reply:
Yeah, I agree. But what other pattern would you recommend for "if the option is set, return that value from the current function, if not, continue"?

Without any "unsafe" operations, it would be as clunky as this (unless I'm missing something):
```
func doSomething(foo fn.Option[bar]) (bar, error) {
    var (
        priorityBar bar
        fooSet bool
    )
    foo.WhenSome(func(b bar) {
        priorityBar = b
        fooSet = true
    })

    if fooSet {
       return priorityBar, nil
    }

    // Find another source for bar...
}
```

Reply:
I think we should add this function to `fn`:
```golang
// Deref safely dereferences a pointer.
// If the pointer is nil, it returns the zero value of type T and an error.
func Deref[T any](ptr *T) (T, error) {
	if ptr == nil {
		var zero T
		return zero, errors.New("nil pointer dereference")
	}
	return *ptr, nil
}
```

It would be useful outside this use case.

And then we can just do:

```golang
func doSomething(foo fn.Option[bar]) (bar, error) {
   if v := foo.UnwrapToPtr(); v != nil {
       return fn.Deref(v)
   }

   // Find another source for bar...
}
```

Eventually we could get to a place where pointer dereference only happens in `Deref`. And it's pretty clear that `Deref` will never dereference a `nil` pointer.

Reply:
Maybe we can also add DerefPanic which just panics when the pointer is nil. That way it's clear in our code what we expect when dereferencing pointers. And we always dereference with either `Deref` or `DerefPanic`.

Reply:
IMO unsafe code is only a problem when it's all over the codebase. If it's just in a few select well though out places it isn't a problem.

Reply:
I agree that we should discuss this. But can we please not block this PR on it?
Do you want me to leave things in and add a TODO or revert to an "ugly" but safe method, then we fix things later on?

Reply:
I've removed the `Unwrap` method from `fn` and solved this in a different way (`UnwrapOr()`).